subsys/mpsl/fem/mpsl_fem.c

```diff
@@ -277,11 +277,22 @@ static int fem_nrf21540_gpio_configure(void)
 					     pdn_gpios));
 #endif
 
+	BUILD_ASSERT(
+	  (CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB == CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTA_PROD) ||
+	  (CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB == CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTB_PROD));
```

Comment:
Having this assertion disallows our customers to use UICR based values as we promise in https://infocenter.nordicsemi.com/pdf/nRF21540_PS_v1.1.1.pdf table 6. Our customers can write UICR values to devices an then solder such prepared nrf21540 to their boards. This does not require to have SPI on target board.

I would recommend to remove this BUILD_ASSERT.

If you have good reason to keep it, please move it after line 283
```c
#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), mode_gpios)
```
This way we would at least enable the user to use UICR-based value with `mode` pin tied to gnd or vcc and not controlled by our SoC.

Reply:
That's a very good catch. Still, the proposed solution would probably not actually cause much trouble in case a customer modifies nRF21540's registers before using it with the supported GPIO driver. After all, it's just a Kconfig option used for determining whether `MODE` pin should be set to high or low state. The customer might assign it a different numeric value than the actual gain used by the hardware only to have `POUTA_UICR` or `POUTB_UICR` chosen through the `MODE` pin. The only problem then is that wrong value of gain is passed through `mpsl_fem_nrf21540_gpio_interface_config_t.pa_gain_db` to MPSL, which might lead to more inconsistencies down the road.

On the other hand, the proposed implementation only handles `MODE` pin correctly when TX gain is set to one of the predefined values. There's no mechanism to handle custom gain provided by the application. That's why in my opinion we should leave the `BUILD_ASSERT`, but only when `mode_gpios` is defined, as @ankuns suggested - that's the currently supported solution. If the user wants to provide their own TX gain - fine, but they should handle the `MODE` pin on their own then.

Reply:
Could we leave the build assert always and allow user to configure the values for POUTA and POUTB?
If the SPI is not yet supported on our end, but customer could "preprogram" FEM before soldering it then it seems like the most proper solution to me.

Reply:
That's kind of workaround

Reply:
@MarekPieta in my opinion that's acceptable, but please then rename `CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTA_PROD` to `CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTA`. The Kconfig would then represent the gain value in POUTA, with the default value equal to POUTA_PROD. If a customer then wishes to set POUTA_SEL and set their custom gain in UICR, that Kconfig symbol would have to be changed through an overlay so that it's different than the default, equal to the custom value. @ankuns what do you think about that?

Reply:
Yes, of course in that case I would need to rename these as you suggest.

Reply:
Is there a good reason to have the `BUILD_ASSERT` before `#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), mode_gpios)` ?
I can't see benefit of asserting if there is no `mode_gpios`. Please explain me this reason.

Reply:
I modified the implementation. The POUTA and POUTB values can be set the by user if he wants to rely on UICR values. The default values of Kconfigs represent production defaults. In that case we may want to verify it no matter if MODE pin is actually set I think (if we tie MODE pin to VCC or GND then still one of the values will be used)